### DISCUSSION TITLE
Addition of a "template" body type

### DISCUSSION THREAD

**Discussion by SummersRemote** [Orange-OpenSource/hurl#2713] [Ideas]
Please forgive me if this is a feature that I have not yet found :)

I am using hurl to load fairly complex test data with lots of inter-data dependencies to a suite of services.  The approach I've been using is to have a separate file for each xml file to be posted (about 150 files) and then use a .hurl file to orchestrate the requests.  After each file is created, I capture some data from the response and use that to create additional data to POST (another additional 50-100 requests).

For example, inside the .hurl file I have a lot of statements like this
```
POST {{host}}/{{endpoint}}
Content-Type: application/xml
Accept: application/xml

file,items/ITEM_A.xml;

[Captures]
ITEM_A-ID: xpath "string(//*[local-name()='id'])"
```

The problem I have is that to use any of the captured values, I must created the request body in the same hurl file.
What I would like is to allow template variables inside of a file-type body so that I could use the captured vars in constructing dynamic requests, but keep the xml in a separate file as these files can be generated by another tool.

I propose adding a new body type called `template` that behaves similarly to `file`, but processes template variables before submitting the request.

**Comment by jcamiel** [comment]
Hi @SummersRemote, thanks for the proposal it reminds me of #1116, we surely can do something about it!

**Comment by fabricereix** [comment]
Yes, this is the #1116 issue.
I'd still would like to see the `file` keyword somewhere to be explicit about the fact that it comes from an external file.
```
file-template,body.json;application/json
```
or
```
file,body.json;application/json;template=true
```
